src/hdr_histogram.h

```diff
@@ -31,7 +31,7 @@ struct hdr_histogram
     double conversion_ratio;
     int32_t counts_len;
     int64_t total_count;
-    int64_t counts[0];
+    int64_t counts[1];
```

Comment:
Do you still get an error if you move the extern "C" that is current below the struct definition to include the struct.  Does the compiler still complain about the zero-length array.

Reply:
Yes, this is actually a C warning, not a C++ one, so you can trigger the warning if you enable -Wpedantic in the top level CMakeLists.txt, and the build of the library gives this error.

https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html discusses what to do about it. The way I read this, the modern approach would be to use int64_t counts[];, but this will not work with earlier C standard (before C99), and given the MS CI build doesn't support virtual, i'm guessing it is not a C99 build.

By making it [1], it will behave exactly the same as before, with the exception that sizeof(hdr_histogram) will be one larger. I don't think this matters though, so I believe it is a safe change (and the unit tests still pass).

There are other uses of [0] arrays in hdr_histogram_log.c, and these do use sizeof(), so i'm leaving these as is. It would be better to fix them, but the code will not be as clear unfortunately (if you need [1] then sizeof() - 1 which is nasty looking.

The change here is in a public interface, hence the need to fix it for users of the library who are being pedantic (which is my use case!).

Reply:
I want to spend a bit more time this week digging into it.  I'll probably go with counts[1], but would like to make sure changing the reported size of the header won't be problematic and whether to so with changing the sizeof logic to use offsetof instead.

Reply:
The other option I am thinking about is just replacing it with a pointer.  One outstanding feature I had meant to implement is growable histograms which requires a separation between the header and data block.